src/providers/oracle/qgsoracledataitems.cpp

```diff
@@ -182,56 +182,50 @@ QVector<QgsDataItem *> QgsOracleConnectionItem::createChildren()
   if ( deferredDelete() )
     return QVector<QgsDataItem *>();
 
-  if ( !mColumnTypeThread )
+  if ( !mColumnTypeTask )
   {
-    mColumnTypeThread = new QgsOracleColumnTypeThread( mName,
+    mColumnTypeTask = new QgsOracleColumnTypeTask( mName,
         QgsOracleConn::restrictToSchema( mName ),
         /* useEstimatedMetadata */ true,
         QgsOracleConn::allowGeometrylessTables( mName ) );
-    mColumnTypeTask = new QgsProxyProgressTask( tr( "Scanning tables for %1" ).arg( mName ) );
-    QgsApplication::taskManager()->addTask( mColumnTypeTask );
 
-    connect( mColumnTypeThread, &QgsOracleColumnTypeThread::setLayerType,
+    connect( mColumnTypeTask, &QgsOracleColumnTypeTask::setLayerType,
              this, &QgsOracleConnectionItem::setLayerType );
-    connect( mColumnTypeThread, &QThread::started, this, &QgsOracleConnectionItem::threadStarted );
-    connect( mColumnTypeThread, &QThread::finished, this, &QgsOracleConnectionItem::threadFinished );
+    connect( mColumnTypeTask, &QgsTask::begun, this, &QgsOracleConnectionItem::taskStarted );
+    connect( mColumnTypeTask, &QgsTask::taskCompleted, this, &QgsOracleConnectionItem::taskFinished );
+    connect( mColumnTypeTask, &QgsTask::taskTerminated, this, &QgsOracleConnectionItem::taskFinished );
 
     if ( QgsOracleRootItem::sMainWindow )
     {
-      connect( mColumnTypeThread, &QgsOracleColumnTypeThread::progress,
-               mColumnTypeTask, [ = ]( int i, int n )
-      {
-        mColumnTypeTask->setProxyProgress( 100.0 * static_cast< double >( i ) / n );
-      } );
-      connect( mColumnTypeThread, SIGNAL( progressMessage( QString ) ),
+      connect( mColumnTypeTask, SIGNAL( progressMessage( QString ) ),
                QgsOracleRootItem::sMainWindow, SLOT( showStatusMessage( QString ) ) );
     }
-  }
 
-  if ( mColumnTypeThread )
-  {
-    mColumnTypeThread->start();
-  }
-  else
-  {
-    setAllAsPopulated();
+    // createChildren is executed in a separate thread that would end with this function scope,
+    // so we move the task to the main thread to be sure there is a running QEventLoop to catch
+    // our signals/events
+    mColumnTypeTask->moveToThread( QCoreApplication::instance()->thread() );
+    QgsApplication::taskManager()->addTask( mColumnTypeTask );
```

Comment:
This isn't safe to do from a background thread. But you can avoid createChildren being run in a thread by adding the `Qgis::BrowserItemCapability::Fast` flag to the item. That flag indicates that createChildren() will return quickly and so is safe to run in the main thread.

Reply:
It's way better indeed, thanks